presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SetFlatteningOptimizer.java

```diff
@@ -85,6 +86,34 @@ public PlanNode visitUnion(UnionNode node, RewriteContext<Boolean> context)
         }
 
         @Override
+        public PlanNode visitIntersect(IntersectNode node, RewriteContext<Boolean> context)
```

Comment:
hmm..not sure how. This optimizer is for combining multiple Set operations into one. For eg: `a interesect b intersect c`, will be converted to one intersect node with sources a,b, c. But in case of `a union b intersect c ` we do not want to combine it into one setOperation node.